firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -124,6 +134,51 @@ public Thread newThread(Runnable r) {
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk();
```

Comment:
Why do it in a blocking way? The least we can do is do it on a thread and post the exception to the caller thread(or UI thread) through its Looper